apps/nextjs-website/src/lib/api.ts

```diff
@@ -1,12 +1,21 @@
 /** This file contains all the functions useful to get data from external resources */
+import { pipe } from 'fp-ts/lib/function';
+import * as E from 'fp-ts/lib/Either';
+import { getNavigation } from './navigation';
+import { Page, makePageListFromNavigation } from './pages';
+import { makeAppEnv } from '@/AppEnv';
+
+// create AppEnv given process env
+const appEnv = pipe(
+  makeAppEnv(process.env),
+  E.getOrElseW((errors) => {
+    // eslint-disable-next-line functional/no-throw-statements
+    throw errors;
+  })
+);
 
 // Return all the pages
-export const getAllPages = async () => ({
-  // for now hardcoded, then retrieved from the cms
-  pages: [
-    { slug: ['a'] },
-    { slug: ['a', 'b'] },
-    { slug: ['a', 'b', 'c'] },
-    { slug: ['a', 'b', 'c', 'd'] },
-  ],
-});
+export const getAllPages = async (): Promise<ReadonlyArray<Page>> => {
+  const navigation = await getNavigation('main-menu', appEnv);
```

Comment:
```suggestion
  const navigation = await getNavigation('main-navigation', appEnv);
```
I would suggest changin 'main-menu' to 'main-navigation' as its the default name for the first navigation menu created or, even better, calling /api/navigation to get the first menu's ID or Slug so as to not have this be an issue.